src/osgEarthDrivers/gdal/ReaderWriterGDAL.cpp

```diff
@@ -733,6 +733,72 @@ build_vrt(std::vector<std::string> &files, ResolutionStrategy resolutionStrategy
     return hVRTDS;
 }
 
+static std::vector<std::tuple<double, double, double, double>>
+build_tilesExtent(std::vector<std::string> &files)
+{
+    GDAL_SCOPED_LOCK;
+
+    int nInputFiles = files.size();
+
+    DatasetProperty* psDatasetProperties =
+            (DatasetProperty*) CPLMalloc(nInputFiles*sizeof(DatasetProperty));
+
+    std::vector<std::tuple<double, double, double, double>> tilesExtent;
+
+    for(int i=0;i<nInputFiles;i++)
+    {
+        const char* dsFileName = files[i].c_str();
+
+        GDALTermProgress( 1.0 * (i+1) / nInputFiles, NULL, NULL);
+
+        GDALDatasetH hDS = GDALOpen(dsFileName, GA_ReadOnly );
+        psDatasetProperties[i].isFileOK = FALSE;
+
+        if (hDS)
+        {
+            const char* proj = GDALGetProjectionRef(hDS);
```

Comment:
If i understand correctly all files are open twice, as well as extracting the projection information.
That's not performant enough. Sorry....
Integrate this in the main 'read' loop and avoid any duplicated work.